src/envoy/http/metadata_exchange/plugin.cc

```diff
@@ -0,0 +1,374 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NULL_PLUGIN
+#include "plugin.h"
```

Comment:
Did you change anything or just a pure copy paste? If it is a pure copy paste, can we do it like what you did for abseil (donwload the envoy repo and just compile with the file)? I think it worth some comments here...

Reply:
Unfortunately that file pulls in envoy/buffer/buffer.h which in turn pulls in a good deal more.
Envoy is not designed to be included piecemeal into other programs AFAICT.

Perhaps we should be using base64 library that is?

Reply:
I thought we decided to place this in `bas64.cc` or `base64.h`

Reply:
Done.